web/core/src/jsMain/kotlin/org/jetbrains/compose/web/css/StyleSheet.kt

```diff
@@ -80,7 +80,8 @@ open class StyleSheet(
 
     @Suppress("EqualsOrHashCode")
     class CSSSelfSelector(var selector: CSSSelector? = null) : CSSSelector() {
-        override fun toString(): String = selector.toString()
+        override fun toString(): String = throw IllegalStateException("You can't concatenate `String + CSSSelector` which contains `self` or `root`. Use `selector(<your string>)` to convert `String` to `CSSSelector` for proper work.")
+        override fun asString(): String = selector?.asString() ?: throw IllegalStateException("You can't instantiate self without actual selector")
```

Comment:
Two question here
1) (just to see a bigger picture if we can not have nullable selector in any practical scenario why should we allow passing nullable selector at all? When we can end up in such situation

2) Even if we'll have null there for any reason - can we just cast  it with !! so that we fail - we'll let user that something is null why we are not expecting it to be null but user will see the standard message they got used to see.

Reply:
1. It's bacause some complicated experimantal scenario with "auto" css, which doesn't know instance of self until instantiation and generating class for it. I just reused this class for this case too.
2. It's more for us, not for users... In Space we have another policy - do not use `!!`, but write custom understandable exceptions